### PR TITLE
Ensure valid git author names passed in signatures

### LINKED ISSUE #5772: Internal Server Error when creating a repository in edge case
I do not have Gitea installed and discovered this very weird bug at https://try.gitea.io, which currently uses af45648.

## Description

1. Create an account
2. Go to your account settings and set your Full Name to `<Test` (it needs to start with a less-than sign)
3. Attempt to create a repository: enter a name and check `Initialize Repository (Adds .gitignore, License and README)` (this is important it will not fail if you do not check it). Click `Create Repository`.
4. Internal Server Error 500

### PR DESCRIPTION
**Pull request by zeripath**
Fix #5772 - Git author names are not allowed to include `\n` `<` or `>` and
must not be empty. Ensure that the name passed in a signature is valid.

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Comment by zeripath** [comment]
This should probably be backported to at least v1.7.

**Comment by adelowo** [comment]
Might need some Tests ????

**Review by techknowlogick — APPROVED** [review]

**Review by techknowlogick — APPROVED** [review]
LGTM. I second what @adelowo days re:tests

File: models/user.go [hunk 1]
```diff
@@ -629,8 +629,18 @@ func (u *User) GetOrganizations(all bool) error {
...
+	trimmed := strings.TrimSpace(u.FullName)
+	if len(trimmed) > 0 {
+		return trimmed
+	}
+	return u.Name
+}
+
+// GitName returns a git safe name
+func (u *User) GitName() string {
+	gitName := strings.TrimSpace(strings.NewReplacer("\n", "", "<", "", ">", "").Replace(u.FullName))
```
Comment (lunny) [inline models/user.go]: Maybe 
```
.Replace(u.DisplayName())
```
is better.

**Review by lunny — CHANGES_REQUESTED** [review]

File: models/user.go (hunk 1 above)
Comment (zeripath) [reply models/user.go]: Hmm, well the only benefit of passing in `u.DisplayName()` (at this point) is that, it's guaranteed to have been whitespace trimmed or be equal to `u.Name`.

You have to `TrimSpace` after removing the bad characters as you could have an initial `<` followed by spaces, or a terminal `<` preceded by spaces. So the `TrimSpace` in `u.DisplayName()` isn't actually useful and AFAICS `u.Name` is guaranteed to be of an appropriate form for git's author name as those characters are not allowed in u.Name and is pretrimmed.

So in the end all display name does is create a string that doesn't get used.

Comment (zeripath) [reply models/user.go]: I have however discovered that `u.Name` isn't guaranteed to not have these characters therefore I've placed a second check on the `u.Name` before declaring the names totally pathological and rendering their name as `"user-$UID"`.

**Comment by zeripath** [comment]
OK tests are up. I've made the test fixtures for user a little more pathological. We should try to make our test fixtures a bit more pathological in general.

File: integrations/api_user_orgs_test.go [hunk 2]
```diff
@@ -9,6 +9,7 @@ import (
 	"net/http"
 	"testing"
 
+	"code.gitea.io/gitea/models"
```
Comment (lafriks) [inline integrations/api_user_orgs_test.go:12]: there should be newline after this line

**Comment by zeripath** [comment]
Hi @lafriks @lunny any more changes needed?

**Comment by lafriks** [comment]
I'm ok with changes, needs @lunny approval

**Review by lunny — APPROVED** [review]

**Pull request merged**